src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -56,59 +52,85 @@ internal class QueueItem<TRequest, TResponse, TRequestContext> : IQueueItem<TReq
         LspServices = lspServices;
 
         MethodName = methodName;
-        Language = language;
 
         var telemetryService = lspServices.GetService<AbstractTelemetryService>();
 
         _requestTelemetryScope = telemetryService?.CreateRequestScope(methodName);
     }
 
-    public static (IQueueItem<TRequestContext>, Task<TResponse>) Create(
+    public static (IQueueItem<TRequestContext>, Task<object?>) Create(
         string methodName,
-        string language,
-        TRequest request,
+        object? serializedRequest,
         ILspServices lspServices,
         ILspLogger logger,
         CancellationToken cancellationToken)
     {
-        var queueItem = new QueueItem<TRequest, TResponse, TRequestContext>(
+        var queueItem = new QueueItem<TRequestContext>(
             methodName,
-            language,
-            request,
+            serializedRequest,
             lspServices,
             logger,
             cancellationToken);
 
         return (queueItem, queueItem._completionSource.Task);
     }
 
-    public async Task<TRequestContext> CreateRequestContextAsync(IMethodHandler handler, CancellationToken cancellationToken)
+    public async Task<TRequestContext> CreateRequestContextAsync<TRequest>(TRequest request, IMethodHandler handler, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
 
         _requestTelemetryScope?.RecordExecutionStart();
 
         var requestContextFactory = LspServices.GetRequiredService<AbstractRequestContextFactory<TRequestContext>>();
-        var context = await requestContextFactory.CreateRequestContextAsync(this, handler, _request, cancellationToken).ConfigureAwait(false);
+        var context = await requestContextFactory.CreateRequestContextAsync(this, handler, request, cancellationToken).ConfigureAwait(false);
         return context;
     }
 
+    public TRequest? TryDeserializeRequest<TRequest>(AbstractLanguageServer<TRequestContext> languageServer, RequestHandlerMetadata requestHandlerMetadata, bool isMutating)
+    {
+        try
+        {
+            return languageServer.DeserializeRequest<TRequest>(SerializedRequest, requestHandlerMetadata);
+        }
+        catch (Exception ex)
+        {
+            // Report the exception to logs / telemetry
+            _requestTelemetryScope?.RecordException(ex);
+            _logger.LogException(ex);
+
+            // Set the task result to the exception
+            _completionSource.TrySetException(ex);
+
+            // End the request - the caller will return immediately if it cannot deserialize.
+            _requestTelemetryScope?.Dispose();
+            _logger.LogEndContext($"{MethodName}");
+
+            // If the request is mutating, bubble the exception out so the queue shutsdown.
+            if (isMutating)
+            {
+                throw;
+            }
+            else
+            {
+                return default;
```

Comment:
ah, this answers teh prior question.  i'd like if the prior docs can explain this.